exporter/datadogexporter/traces_exporter_test.go

```diff
@@ -0,0 +1,315 @@
+
+	if err != nil {
+		http.Error(rw, err.Error(), http.StatusInternalServerError)
+		assert.NoError(t, err, "http server received malformed trace payload")
+		return
+	}
+
+	defer req.Body.Close()
+
+	if marshallErr := proto.Unmarshal(b, &traceData); marshallErr != nil {
+		http.Error(rw, marshallErr.Error(), http.StatusInternalServerError)
+		assert.NoError(t, marshallErr, "http server received malformed trace payload")
+		return
+	}
+
+	assert.NotNil(t, traceData.Env)
+	assert.NotNil(t, traceData.HostName)
+	assert.NotNil(t, traceData.Traces)
+}
+
+func testJSONTraceStatsPayload(t *testing.T, rw http.ResponseWriter, req *http.Request) {
+	var statsData stats.Payload
+
+	gz, err := gzip.NewReader(req.Body)
+	if err != nil {
+		http.Error(rw, err.Error(), http.StatusInternalServerError)
+		require.NoError(t, err, "http server received malformed stats payload")
+		return
+	}
+
+	defer req.Body.Close()
+	defer gz.Close()
+
+	statsBytes, err := ioutil.ReadAll(gz)
+	if err != nil {
+		http.Error(rw, err.Error(), http.StatusInternalServerError)
+		require.NoError(t, err, "http server received malformed stats payload")
+		return
+	}
+
+	if marshallErr := json.Unmarshal(statsBytes, &statsData); marshallErr != nil {
+		http.Error(rw, marshallErr.Error(), http.StatusInternalServerError)
+		require.NoError(t, marshallErr, "http server received malformed stats payload")
+		return
+	}
+
+	assert.NotNil(t, statsData.Env)
+	assert.NotNil(t, statsData.HostName)
+	assert.NotNil(t, statsData.Stats)
+}
+
+func TestNewTraceExporter(t *testing.T) {
+	cfg := &config.Config{}
+	cfg.API.Key = "ddog_32_characters_long_api_key1"
+	logger := zap.NewNop()
+
+	// The client should have been created correctly
+	exp, err := newTraceExporter(logger, cfg)
+	assert.NoError(t, err)
+	assert.NotNil(t, exp)
+}
+
+func TestPushTraceData(t *testing.T) {
+	server := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		assert.Equal(t, "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", req.Header.Get("DD-Api-Key"))
+		rw.WriteHeader(http.StatusAccepted)
+	}))
+
+	defer server.Close()
+	cfg := &config.Config{
+		API: config.APIConfig{
+			Key: "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+		},
+		TagsConfig: config.TagsConfig{
+			Hostname: "test_host",
+			Env:      "test_env",
+			Tags:     []string{"key:val"},
+		},
+		Traces: config.TracesConfig{
+			SampleRate: 1,
+			TCPAddr: confignet.TCPAddr{
+				Endpoint: server.URL,
+			},
+		},
+	}
+	logger := zap.NewNop()
+
+	exp, err := newTraceExporter(logger, cfg)
+
+	assert.NoError(t, err)
+
+	tracesLength, err := exp.pushTraceData(context.Background(), func() pdata.Traces {
+		traces := pdata.NewTraces()
+		resourceSpans := traces.ResourceSpans()
+		resourceSpans.Resize(1)
+		resourceSpans.At(0).InitEmpty()
+		resourceSpans.At(0).InstrumentationLibrarySpans().Resize(1)
+		resourceSpans.At(0).InstrumentationLibrarySpans().At(0).Spans().Resize(1)
+		return traces
+	}())
+
+	assert.Nil(t, err)
```

Comment:
If you are updating this PR for some other reason, update also calls like this to use `NoError`.

Reply:
👍 Updated